src/main/java/uk/gov/hmcts/reform/idam/web/UserController.java

```diff
@@ -253,25 +257,29 @@ public String selfRegisterUser(@ModelAttribute("selfRegisterCommand") @Validated
      * @should return expiredtoken view if HttpClientErrorException occurs and http status is 400 and token is invalid
      * @should return redirect expiredtoken page if selfRegisterUser service throws HttpClientErrorException and Http code is 404
      */
-    @RequestMapping(path = "/activate", method = RequestMethod.POST)
-    public String activateUser(@RequestParam("token") String token, @RequestParam("code") String code, @RequestParam("password1") String password1, @RequestParam("password2") String password2,
-                               final Map<String, Object> model) throws IOException {
+    @PostMapping("/activate")
+    public ModelAndView activateUser(@RequestParam("token") String token, @RequestParam("code") String code,
+                                     @RequestParam("password1") String password1, @RequestParam("password2") String password2,
+                                     final Map<String, Object> model) throws IOException {
         model.put("token", token);
         model.put("code", code);
         try {
             if (validationService.validatePassword(password1, password2, model)) {
                 String activation = "{\"token\":\"" + token + "\",\"code\":\"" + code + "\",\"password\":\"" + password1 + "\"}";
                 ResponseEntity<String> response = spiService.activateUser(activation);
                 String redirectUri = getRedirectUri(response.getBody());
+                // don't expose parameters other than the url to a GET request
+                Map<String, Object> successModel = new HashMap<>();
                 if (redirectUri != null) {
-                    model.put("redirectUri", redirectUri);
+                    successModel.put("redirectUri", redirectUri);
                 }
 
-                return "useractivated";
+                return new ModelAndView("redirect:useractivated", successModel);
             }
         } catch (HttpClientErrorException e) {
             if (e.getStatusCode() == HttpStatus.NOT_FOUND) {
-                return "redirect:expiredtoken";
+                // don't expose the token in the error page
+                return new ModelAndView("redirect:expiredtoken", ImmutableMap.of());
```

Comment:
is it really worth adding an import just for this?

Reply:
you're correct, the method accepts `null`